src/spatialdata/_core/concatenate.py

```diff
@@ -108,16 +114,29 @@ def concatenate(
     assert isinstance(sdatas, list), "sdatas must be a list"
     assert len(sdatas) > 0, "sdatas must be a non-empty list"
 
-    merged_table = _concatenate_tables(
-        [sdata.table for sdata in sdatas if sdata.table is not None], region_key, instance_key, **kwargs
-    )
+    if not concatenate_tables:
+        merged_tables = {**{k: v for sdata in sdatas for k, v in sdata.tables.items()}}
+        if len(merged_tables) != np.sum([len(sdata.tables) for sdata in sdatas]):
+            raise KeyError(
+                "Tables must have unique names across the SpatialData objects to concatenate unless"
+                " concatenate_tables is set to True."
+            )
+    else:
+        common_keys = find_common_table_keys(sdatas)
+        merged_tables = {}
+        for sdata in sdatas:
+            for k, v in sdata.tables.items():
+                if k in common_keys and merged_tables.get(k) is not None:
+                    merged_tables[k] = _concatenate_tables([merged_tables[k], v], region_key, instance_key, **kwargs)
+                else:
+                    merged_tables[k] = v
```

Comment:
We could reduce the number of _concatenate_tables() being called by calling it once for each unique table name for which there are > 1 tables.

Reply:
This is essentially what is happening here using `common_keys`. The `common_keys` are the table_names for which in the list of SpatialData objects multiple tables exist.

Reply:
Marking this as resolved for now as I think the functionality as described is already there.